src/gui/maptools/qgsmaptoolcapture.cpp

```diff
@@ -154,6 +154,16 @@ void QgsMapToolCapture::currentLayerChanged( QgsMapLayer *layer )
     return;
   }
 
+  if ( vlayer->isSpatial() )
```

Comment:
This is not the correct approach.
This map tool is a base class and is used to capture even if there is no current layer.
This is breaking this scenerio.